src/main/webapp/components/workspaces/workspaces.js

```diff
@@ -59,6 +68,181 @@ const workspaceById = gql`
     }
   }
 `
+const queryAttributes = gql`
+  fragment QueryAttributes on MetacardAttributes {
+    title
+    cql
+    metacard_tags
+    metacard_type
+  }
+`
+
+const useCreateQuery = workspaceId => {
+  const [queryId, setQueryId] = React.useState('')
+  const mutation = gql`
+    mutation CreateQuery($attrs: MetacardAttributesInput!) {
+      createMetacard(attrs: $attrs) {
+        id: id
+        ...QueryAttributes
+      }
+    }
+    ${queryAttributes}
+  `
+
+  const saveMutation = gql`
+    mutation SaveWorkspace($id: ID!, $attributes: Json!) {
+      saveMetacardFromJson(id: $id, attributes: $attributes) {
+        ...WorkspaceAttributes
+      }
+    }
+    ${workspaceAttributes}
+  `
+
+  const [save] = useMutation(saveMutation)
+
+  const [create] = useMutation(mutation, {
+    onCompleted: data => {
+      save({
+        variables: {
+          id: workspaceId,
+          attributes: {
+            queries: [data.createMetacard.id],
+            metacard_type: 'workspace',
+          },
+        },
+        refetchQueries: ['WorkspaceById'],
+      }),
+        setQueryId(data.createMetacard.id)
```

Comment:
You probably want to wait to set the query id until after it's been added to the workspace.
```suggestion
    onCompleted: async data => {
      await save({
        variables: {
          id: workspaceId,
          attributes: {
            queries: [data.createMetacard.id],
            metacard_type: 'workspace',
          },
        },
        refetchQueries: ['WorkspaceById'],
      }),
      setQueryId(data.createMetacard.id)
```